File: benchmarks/strings/string_manipulation.js
// String manipulation - JavaScript
// Tests native string methods

const text = "The Quick Brown Fox Jumps Over The Lazy Dog";
let count = 0;

for (let i = 0; i < 10000; i++) {
    const upperText = text.toUpperCase();
    const lowerText = text.toLowerCase();
    const pos = lowerText.indexOf("fox");
    const sliced = upperText.slice(0, 10);
    count += sliced.length + pos;
}

console.log(count);

File: benchmarks/strings/string_manipulation.py
# String manipulation - Python
# Tests native string methods

text = "The Quick Brown Fox Jumps Over The Lazy Dog"
count = 0

for i in range(10000):
    upper_text = text.upper()
    lower_text = text.lower()
    pos = lower_text.find("fox")
    sliced = upper_text[0:10]
    count += len(sliced) + pos

print(count)